grammars.nix

```diff
@@ -40,24 +33,13 @@
   grammarsToUse = builtins.filter useGrammar languagesConfig.grammar;
   gitGrammars = builtins.filter isGitGrammar grammarsToUse;
   buildGrammar = grammar: let
-    gh = toGitHubFetcher grammar.source.git;
-    sourceGit = builtins.fetchTree {
+    source = builtins.fetchTree {
       type = "git";
       url = grammar.source.git;
       rev = grammar.source.rev;
       ref = grammar.source.ref or "HEAD";
```

Comment:
The `nix build` fails for me on this branch with Nix 2.18.1. A few grammars in `languages.toml` use commits that aren't from their HEAD, for example tree-sitter-groovy is from the `gh-pages` branch:

```
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'helix-term-23.10.0'
         whose name attribute is located at /nix/store/bjvqq8c79dbi59g7xzcc6lhl0f19m3d7-source/pkgs/stdenv/generic/make-derivation.nix:353:7

       … while evaluating attribute 'buildCommand' of derivation 'helix-term-23.10.0'

         at /nix/store/bjvqq8c79dbi59g7xzcc6lhl0f19m3d7-source/pkgs/build-support/trivial-builders/default.nix:98:16:

           97|         enableParallelBuilding = true;
           98|         inherit buildCommand name;
             |                ^
           99|         passAsFile = [ "buildCommand" ]

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: Cannot find Git revision '7e023227f46fee428b16a0288eeb0f65ee2523ec' in ref 'HEAD' of repository 'https://github.com/Decodetalkers/tree-sitter-groovy'! Please make sure that the rev exists on the ref you've specified or add allRefs = true; to fetchGit.
```

I believe this needs to be

```suggestion
      allRefs = true;
```

instead, or we need to add `ref` to all grammars that don't use their main branch for committing tree-sitter outputs. (We should use `allRefs` instead because not all language-support contributors use Nix so they won't know to update the `ref`.)